libethereum/BlockChainSync.cpp

```diff
@@ -463,6 +463,11 @@ void BlockChainSync::onPeerBlockHeaders(std::shared_ptr<EthereumPeer> _peer, RLP
 					m_downloadingHeaders.erase(blockNumber - 1);
 					removeItem(m_headers, blockNumber - 1);
 					removeItem(m_bodies, blockNumber - 1);
+					m_lastImportedBlock--;
+					m_highestBlock = m_lastImportedBlock;
```

Comment:
Hm, not totally sure about this change. We do not have `m_lastImportedBlock == blockNumber - 1` in all cases in this if, furthemore, should we actually delete a block that we already imported? Can you check back with arkadiy, perhaps?